### ISSUE TITLE
"Title", "Genre" and "Rating" fields are required but screen reader is not announcing as required: A11y_ASP.NET_Movies_Create Movies_ScreenReader

### ISSUE THREAD

**Issue by PoojaNamde** [dotnet/aspnetcore#56283]
GitHub Tags:
#A11yMAS; #A11yTCS; #A11ySev2; #BM_ASP.NET_Web_Jun2024; #.NETCore; #WCAG1.3.1; #DesktopWeb; #Win11; #FTP; #Narrator; #NVDA; #Element:EditFields; #ChromiumEdge; #Chrome;

### Environment Details:
App name: ASP.NET
Window Version: Win 11
OS Build: 22621.3085
Screen readers: Narrator, NVDA 2023.3.3

### Repro Steps:

1. Turn on Narrator/NVDA.
2. Download the "BlazorWebApp" zip file and run .exe file.
3. Command prompt will open, TAB to "http://localhost:5000/" link and press CTRL + ENTER key.
4. TAB to "Movies" tab and hit ENTER key. TAB to "Create New" button and hit ENTER key.
5. TAB to "Title", "Genre" and "Rating" fields and observe that Narrator screen reader is not announcing these fields as required

### Actual Result:
"Title", "Genre" and "Rating" fields are required but Narrator screen reader is not announcing as required. Also * is not provided for required fields.

**Similar issue is also observed with Chrome + NVDA.**

### Expected Result:
"Title", "Genre" and "Rating" fields are required hence Narrator screen reader should be announcing as required. Also * should be provided for required fields.

### User Impact:
Screen reader users will not be able to get information about the required fields and will miss out the information.

### Attachment
[56283_A11y_ASP.NET_Movies_Create Movies_NVDA.webm](https://github.com/dotnet/aspnetcore/assets/90901519/0afa7e61-1c9d-4c24-b509-376d45471898)

[56283_A11y_ASP.NET_Movies_Create Movies_Narrator.webm](https://github.com/dotnet/aspnetcore/assets/90901519/4925cb49-8f22-4a76-9c63-d511a682d062)

**Comment by Yash14j** [comment]
GithubTags:#Rev:yaja;

**Comment by PoojaNamde** [comment]
#Regressed:09-20-24;

Verified the bug in the environment http://localhost:5000/movies/create
still able to repro it. hence reactivating it.
Please find repro attachment for reference.

![image](https://github.com/user-attachments/assets/fa364751-2c3c-4a10-860f-310350616112)

**Comment by PoojaNamde** [comment]
#Regressed:11-12-24;

Verified the issue in latest environment. Issue is not fixed as screen reader is not announcing the fields as required. Please find below screenshot for reference.


![Image](https://github.com/user-attachments/assets/1ff59d07-cc80-4505-873d-f91b5504f8bf)

**Comment by PoojaNamde** [comment]
#Regressed:11-15-24;

Verified the issue in latest environment. Issue is partially fixed as screen reader is announcing the fields as required but the visual * asterisk is not provided for required fields.

**Comment by PoojaNamde** [comment]
#Closed; #Regressed:12-06-24;

Verified the issue in latest env, issue is fixed hence closing the bug.

![Image](https://github.com/user-attachments/assets/b5f4dad0-5d70-4006-9d86-513704db58cd)
